lua/entities/gmod_wire_expression2/core/custom/wiring.lua

```diff
@@ -108,7 +108,17 @@ __e2setcost(5)
 e2function wirelink entity:wirelink()
 	if not IsValid(this) then return self:throw("Invalid entity!", nil) end
 	if not isOwner(self, this) then return self:throw("You do not own this entity!", nil) end
-	
+
+	return this
+end
+
+e2function wirelink entity:createWirelink()
```

Comment:
This doesn't fix #2940. You need to revert the change. We don't break backwards compatibility for literally no reason. This is just saving a one liner. My idea was to deprecate this, then add `e:getWirelink()` for the new behavior.

Reply:
The wirelink being created isn't even the problem, it's the valid check on wirelinks expecting an `extended` field on the entity to be set, because wirelinks *have* to be extended for no reason.